src/aks-preview/azext_aks_preview/_helpers.py

```diff
@@ -92,3 +95,49 @@ def _trim_fqdn_name_containing_hcp(normalized_fqdn: str) -> str:
     if len(storage_name_without_hcp) > CONST_CONTAINER_NAME_MAX_LENGTH:
         storage_name_without_hcp = storage_name_without_hcp[:CONST_CONTAINER_NAME_MAX_LENGTH]
     return storage_name_without_hcp.rstrip('-')
+
+
+def _fuzzy_match(query, arr):
+    """
+    will compare all elements in @arr against the @query to see if they are similar
+
+    similar implies one is a substring of the other or the two words are 1 change apart
+
+    Ex. bird and bord are similar
+    Ex. bird and birdwaj are similar
+    Ex. bird and bead are not similar
+    """
+    def similar_word(a, b):
+        a_len = len(a)
+        b_len = len(b)
+        if a_len > b_len:  # @a should always be the shorter string
+            return similar_word(b, a)
+        if a == b:
```

Comment:
Nit: please do this on one line

Reply:
The lengths and equality can't be checked in the same line. They have different outcomes.

Reply:
Sorry I mean the two parts that return false - somehow the part that I highlighted seems off. But I think there are two if checks back to back that return false. Just suggesting those become if check1  or check2 --> return false